includes/Services/PluginService.php

```diff
@@ -87,6 +91,73 @@ public static function activate_init_plugins() {
 			);
 		}
 
+		foreach ( $site_features_unselected as $init_plugin ) {
+			$init_plugin_type = PluginInstaller::get_plugin_type( $init_plugin['slug'] );
+			$init_plugin_path = PluginInstaller::get_plugin_path( $init_plugin['slug'], $init_plugin_type );
+			// Checks if a plugin with the given slug and activation criteria already exists.
+			if ( PluginInstaller::is_plugin_installed( $init_plugin_path ) ) {
+					// Add a new PluginDeactivationTask to the Plugin Deactivation queue.
+					PluginDeactivationTaskManager::add_to_queue(
+						new PluginDeactivationTask(
+							$init_plugin['slug']
+						)
+					);
+					continue;
+			}
+
+			PluginInstallTaskManager::add_to_queue(
+				new PluginInstallTask(
+					$init_plugin['slug'],
+					false,
+					isset( $init_plugin['priority'] ) ? $init_plugin['priority'] : 0
+				)
+			);
+		}
+
 		return true;
 	}
+
+	/**
+	 * Sets up a Transient to activate plugins and filter active_plugins
+	 *
+	 * @return boolean
+	 */
+	public static function configure_activation_transient() {
+		global $pagenow;
+		global $current_screen;
+
+		switch ( $pagenow ) {
+			case 'admin.php':
+				self::activate_init_plugins();
+				if ( ! empty( get_transient( Options::get_option_name( 'active_plugins', true ) ) ) ) {
+					delete_transient( Options::get_option_name( 'active_plugins', true ) );
+				}
+				break;
+			case 'plugins.php':
+				self::activate_init_plugins();
+				if ( ! empty( get_transient( Options::get_option_name( 'active_plugins', true ) ) ) ) {
+					delete_transient( Options::get_option_name( 'active_plugins', true ) );
+				}
+				break;
+			case 'index.php':
+				// If the page is nfd-onboarding
+				if ( isset( $_GET['page'] ) && WP_ADMIN::$slug === $_GET['page'] ) {
+					if ( empty( get_transient( Options::get_option_name( 'active_plugins', true ) ) ) ) {
```

Comment:
```suggestion
					if ( empty( get_transient( Options::get_option_name( 'active_plugins' ) ) ) ) {
```
Since default is true anyways. Similarly for other instances of this